supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesCacheHitCopyingActor.scala

```diff
@@ -1,71 +1,104 @@
+        case error: RuntimeException => Future(FailedNonRetryableResponse(jobDescriptor.key, error, Option(-1)))
+      }
     }
 
-    val gcsFileSystem = workflowPaths.gcsFileSystemWithUserAuth
+      def copyFiles(sourceCallRootPath: Path) = {
+        seqOfSimpletons ++ jobDetritusFiles foreach {
+          case WdlValueSimpleton(key, wdlFile: WdlFile) =>
+            copyCachedCallOutputFiles(wdlFile.value, sourceCallRootPath)
+          case (fileName: String, filePath: String) if fileName != jesCallPaths.callRootPathKey =>
+            copyCachedCallOutputFiles(filePath, sourceCallRootPath)
+          case _ =>
+            val error = new RuntimeException(s"Unable to copy cached job outputs for ${jobDescriptor.call.fullyQualifiedName}")
+            FailedNonRetryableResponse(jobDescriptor.key, error, Option(-1))
+        }
+      }
+
+      def copyCachedCallOutputFiles(sourceString: String, srcContextPath: Path): Unit = {
+        val sourcePath = gcsFileSystem.getPath(sourceString)
+        PathCopier.copy(srcContextPath, sourcePath, destinationCallRootPath)
+      }
 
-    def copyFilesInGcs() = {
-      val stringFilePaths = cachedJobOutputs map { case (lqn, jobOutput) => lqn -> jobOutput.wdlValue.valueString }
 
-      stringFilePaths foreach {
-        case (lqn, filePath) => if (filePath.isGcsUrl)
-          copyCachedCallOutputFiles(filePath)
+      def tellMetadata(key: String, value: Any): Unit = {
+        val event = metadataEvent(key, value)
+        val putMetadataAction = PutMetadataAction(event)
+        serviceRegistryActor ! putMetadataAction
       }
-    }
 
-    //nested array output copying? or more complicated file structures
-    def copyCachedCallOutputFiles(sourcePath: String): Unit = {
-      val sourceCallDirectory = gcsFileSystem.getPath(sourcePath)
-      // suspicious that this is unused
-      val buildCallDestinationDirectory = gcsFileSystem.getPathAsDirectory(callRootPath.getParent.toString)
+      def metadataEvent(key: String, value: Any) = {
+        val metadataValue = MetadataValue(value)
+        MetadataEvent(metadataKey(key), metadataValue)
+      }
 
-      PathCopier.copy(callRootPath, sourceCallDirectory.getParent, callRootParent)
-    }
+      def metadataKey(key: String) = MetadataKey(workflowId, Option(metadataJobKey), key)
+
+      def updateMetadata = {
+        tellMetadata(CallMetadataKeys.CallRoot, destinationCallRootPath)
+        tellMetadata(JesMetadataKeys.GoogleProject, jesAttributes.project)
+        tellMetadata(JesMetadataKeys.ExecutionBucket, jesAttributes.executionBucket)
+        tellMetadata(JesMetadataKeys.EndpointUrl, jesAttributes.endpointUrl)
+      }
+
+
+    def createJobOutputs: JobOutputs = WdlValueBuilder.toJobOutputs(jobDescriptor.call.task.outputs, seqOfSimpletons)
 
-    copyFilesInGcs()
-    Future(SucceededResponse(jobDescriptor.key, Option(0), cachedJobOutputs))
+    cacheIfPossible
```

Comment:
After all my previous comments, I wouldn't mind if you just inline'd `cacheIfPossible` here, it's a little tough to follow the logic flow here